scripts/types_generator/schema_transformer.py

```diff
@@ -65,14 +66,29 @@ def _remove_compositions(schema: Any, *, path="", delimiter="."):
 
 
 def path_to_schema_name(path: str, delimiter: str) -> str:
+    """
+    Example::
+
+        /users/{id}/do-something -> userIdDoSomething
+    """
+
     path_parts = path.replace("/v2/", "").strip(delimiter).split("/")
     path = delimiter.join(part.rstrip("s") for part in path_parts)  # make singular
-    return re.sub(
-        r"\{(.*?)}",
-        lambda m: from_camel_case_to_snake_case(m.groups()[0]),
-        path,
-        flags=re.MULTILINE,
-    )
+
+    def _process_parameters(input: str):
+        return re.sub(
+            r"\{(.*?)}",
+            lambda m: from_camel_case_to_snake_case(m.group(1)),
+            input,
+            flags=re.MULTILINE,
+        )
+
+    def _process_hyphens(input: str):
```

Comment:
Maybe we should stop calling things `input` when it's not necessary - it shadows the builtin `input` function.

It makes sense in the user-facing parameters, but here I'd call it `value` or something else :D